### ISSUE TITLE
Auto-regeneration not working, even when forced

### ISSUE THREAD

**Issue by glennreyes** [jekyll/jekyll#4831]
###### What version of Jekyll are you using (`jekyll -v`)?
jekyll 3.0.4

###### What operating system are you using?
OS X 10.11.4

###### What did you do?
1. Setup [Jekyll GitHub Pages](https://help.github.com/articles/setting-up-your-github-pages-site-locally-with-jekyll/)
2. Ran ```bundle exec jekyll serve``` as in the [guide](https://help.github.com/articles/setting-up-your-github-pages-site-locally-with-jekyll/)

###### What did you expect to see?

Auto-regeneration to be enabled like [here](https://help.github.com/articles/setting-up-your-github-pages-site-locally-with-jekyll/#step-4-build-your-local-jekyll-site).

###### What did you see instead?

Auto-regeneration disabled with following output (__note the deprecation notes__):
```sh
Configuration file: /Users/glennreyes/p/my-jekyll-site-project-name/_config.yml
       Deprecation: Auto-regeneration can no longer be set from your configuration file(s). Use the --[no-]watch/-w command-line option instead.
            Source: /Users/glennreyes/p/my-jekyll-site-project-name
       Destination: /Users/glennreyes/p/my-jekyll-site-project-name/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
                    done in 0.283 seconds.
 Auto-regeneration: disabled. Use --watch to enable.
Configuration file: /Users/glennreyes/p/my-jekyll-site-project-name/_config.yml
       Deprecation: Auto-regeneration can no longer be set from your configuration file(s). Use the --[no-]watch/-w command-line option instead.
    Server address: http://127.0.0.1:4000/
  Server running... press ctrl-c to stop.
```

Same output when I run:
```bundle exec jekyll serve --watch```
```bundle exec jekyll serve --w```
```jekyll serve --watch```
```jekyll serve --w```

That is my ```gem list```:
```
*** LOCAL GEMS ***

activesupport (4.2.6)
addressable (2.4.0)
bigdecimal (1.2.8)
bundle (0.0.1)
bundler (1.11.2)
coffee-script (2.4.1)
coffee-script-source (1.10.0)
colorator (0.1)
concurrent-ruby (1.0.1)
did_you_mean (1.0.0)
ethon (0.8.1)
execjs (2.6.0)
faraday (0.9.2)
fastimage (2.0.0)
ffi (1.9.10)
gemoji (2.1.0)
github-pages (75)
github-pages-health-check (1.1.1, 1.1.0)
html-pipeline (2.4.0)
i18n (0.7.0)
io-console (0.4.5)
jekyll (3.0.4)
jekyll-coffeescript (1.0.1)
jekyll-feed (0.5.1)
jekyll-gist (1.4.0)
jekyll-github-metadata (1.11.1)
jekyll-mentions (1.1.2)
jekyll-paginate (1.1.0)
jekyll-redirect-from (0.10.0)
jekyll-sass-converter (1.4.0, 1.3.0)
jekyll-seo-tag (1.3.3)
jekyll-sitemap (0.10.0)
jekyll-textile-converter (0.1.0)
jekyll-watch (1.3.1)
jemoji (0.6.2)
json (1.8.3)
kramdown (1.10.0)
liquid (3.0.6)
listen (3.0.6)
mercenary (0.3.6)
mini_portile2 (2.1.0, 2.0.0)
minitest (5.8.4, 5.8.3)
multipart-post (2.0.0)
net-dns (0.8.0)
net-telnet (0.1.1)
nokogiri (1.6.7.2)
octokit (4.3.0)
octopress-autoprefixer (2.0.0)
power_assert (0.2.7, 0.2.6)
psych (2.0.17)
public_suffix (1.5.3)
rack (1.6.4)
rake (11.1.2, 10.4.2)
rb-fsevent (0.9.7)
rb-inotify (0.9.7)
rdiscount (2.1.8)
rdoc (4.2.2, 4.2.1)
redcarpet (3.3.4, 3.3.3)
RedCloth (4.2.9)
rouge (1.10.1)
safe_yaml (1.0.4)
sass (3.4.22)
sawyer (0.7.0)
sprockets (3.6.0)
sprockets-helpers (1.2.1)
terminal-table (1.5.2)
test-unit (3.1.8, 3.1.5)
thread_safe (0.3.5)
tilt (2.0.2)
typhoeus (1.0.1, 0.8.0)
tzinfo (1.2.2)
```

Can somebody explain what I am doing wrong?

**Comment by pathawks** [comment]
Duplicate of #4830

**Comment by perlun** [comment]
#4830 was closed, but this is indeed a real problem in Jekyll. The `github-pages` gem just happens to make it visible.

[This issue](https://github.com/github/pages-gem/issues/280) has more details, shamelessly quoting:

> This doesn't seem to be a problem with github-pages, but a problem with Jekyll 3.0.4's Configuration.from. #backwards_compatibilize is destructive and can only be called once on a configuration.

Please, leave this issue open until it's fixed for good. If someone can manage to make a failing `spec`, I'm quite sure we can sort it out.

**Comment by lpil** [comment]
Is there a recommended work-around until this bug gets fixed?

**Comment by envygeeks** [comment]
@lpil revert to `github-pages` version `72`

**Comment by lpil** [comment]
Thank you.

**Comment by dotps1** [comment]
@envygeeks I'm a little new to ruby/Jekyll, I am using bundler as well, can you explain how to revert that gem to version 72?

**Comment by glennreyes** [comment]
@dotps1 It's pretty well explained [here](https://github.com/jekyll/jekyll/issues/4830#issuecomment-213937107) at #4830

**Comment by perlun** [comment]
Tested with latest `master` code; worked there. So this is _specifically_ an issue with v3.0.4.

@parkr - someone suggested that the `Configuration#from` method is the problem here, or more specifically the method it calls: `backwards_compatibilize`, which is supposedly destructive. (can't comment on that since it _looks_ sane to me. :smirk:)

I note that `Configuration#from` doesn't exist at all in the latest `master` code. Is that something only on the 3.0 track that is broken in some way?

**Comment by envygeeks** [comment]
@perlun yes.

**Comment by parkr** [comment]
Fixed by #4846.

**Comment by Izhaki** [comment]
Well... with 3.0.5 when you run `bundle exec jekyll serve`

You get:
    Could not find i18n-0.7.0 in any of the sources
    Run `bundle install` to install missing gems. 

And not matter what I've tried (and having spent 3 hours googling and trying different solutions), this error simply won't go away.

May have nothing to do with jekyll. But I'm giving up.

**Comment by perlun** [comment]
@Izhaki - did you try with `bundle install`? What does it give you?

**Comment by Izhaki** [comment]
@perlun 

I did, it just runs throwing no errors. Same for `bundle update` or `bundle pack`, or `bundle install --path vendor/cache`.

This error is thrown with any `bundle exec`.

The gem is clearly there, but it just can't find it for some odd reason.

**Comment by perlun** [comment]
Which version of Ruby do you use? Do you use RVM?

**Comment by pathawks** [comment]
@Izhaki @perlun I am sorry you are having trouble, but your problem does not appear to be related to this issue.

For general troubleshooting, you will have better luck if you post your question at https://talk.jekyllrb.com

**Comment by Izhaki** [comment]
@perlun https://talk.jekyllrb.com/t/could-not-find-i18n-0-7-0-in-any-of-the-sources/2310

**Comment by tomhamiltonstubber** [comment]
This is fixed in jekyll 3.0.5 (github-pages 78) for me.

**Comment by glennreyes** [comment]
Great! I think we can close this issue now.
